cli/src/pcluster/cluster_model.py

```diff
@@ -115,6 +116,62 @@ def _ec2_run_instance(self, pcluster_config, **kwargs):  # noqa: C901 FIXME!!!
                     "Please double check your cluster configuration.\n{1}".format(kwargs["InstanceType"], message)
                 )
 
+    def _get_official_image_id(self, os, architecture):
+        """Return the id of the current official image, for the provided os-architecture combination."""
+        ami_id = None
+        try:
+            images = (
+                boto3.client("ec2")
+                .describe_images(
+                    Filters=[
+                        {
+                            "Name": "name",
+                            "Values": ["{0}*".format(self._get_official_image_name_prefix(os, architecture))],
+                        },
+                        {"Name": "owner-alias", "Values": ["amazon"]},
+                    ],
+                )
+                .get("Images")
+            )
+            if images:
+                ami_id = images[0].get("ImageId")
+        except ClientError as e:
+            raise Exception(
+                "Unable to retrieve official image id for base_os='{0}' and architecture='{1}': {2}".format(
+                    os, architecture, e.response.get("Error").get("Message")
+                )
+            )
+        if not ami_id:
+            raise Exception(
+                "No official image id found for base_os='{0}' and architecture='{1}'".format(os, architecture)
+            )
+
+        return ami_id
+
+    def _get_official_image_name_prefix(self, os, architecture):
+        """Return the prefix of the current official image, for the provided os-architecture combination."""
+        suffixes = {
+            "alinux": "amzn-hvm",
+            "alinux2": "amzn2-hvm",
+            "centos7": "centos7-hvm",
+            "centos8": "centos8-hvm",
+            "ubuntu1604": "ubuntu-1604-lts-hvm",
+            "ubuntu1804": "ubuntu-1804-lts-hvm",
+        }
+        return "aws-parallelcluster-{version}-{suffix}-{arch}".format(
+            version=get_installed_version(), suffix=suffixes[os], arch=architecture
+        )
+
+    def _get_cluster_ami_id(self, pcluster_config):
+        """Get the image id of the cluster."""
+        os = pcluster_config.get_section("cluster").get_param_value("base_os")
+        architecture = pcluster_config.get_section("cluster").get_param_value("architecture")
+        custom_ami = pcluster_config.get_section("cluster").get_param_value("custom_ami")
+        try:
+            return custom_ami if custom_ami else self._get_official_image_id(os, architecture)
+        except Exception as e:
+            pcluster_config.error("Error when resolving cluster ami id: {0}".format(e))
```

Comment:
In this case there is no return value. What happens in the caller?

Reply:
Added return `None` in that case